src/v/cluster/controller_backend.cc

```diff
@@ -350,15 +642,12 @@ ss::future<std::error_code> controller_backend::create_partition(
 }
 
 ss::future<std::error_code>
-controller_backend::delete_partition(model::ntp ntp, model::revision_id rev) {
+controller_backend::delete_partition(model::ntp ntp, model::revision_id) {
     auto part = _partition_manager.local().get(ntp);
     if (unlikely(part.get() == nullptr)) {
         return ss::make_ready_future<std::error_code>(errc::success);
     }
-    // partition was already recreated with greater rev, do nothing
-    if (unlikely(part->get_revision_id() > rev)) {
```

Comment:
interesting. is this condition prevented with some other mechanism? it seems like it is a valid condition to be guarded against?

Reply:
I've missed it, sorry. Brought it back